LanguageFeatures/Patterns/constant_A03_t02.dart

```diff
@@ -23,87 +23,80 @@
 /// literals explicitly marked const. Likewise with set and map literals versus
 /// map patterns.
 ///
-/// @description Check list, map and set literals with type argument specified
-/// in a constant patterns
+/// @description Check list, map and set literals in a constant patterns. Test
+/// if-case statement
 /// @author sgrekhov22@gmail.com
 
 // SharedOptions=--enable-experiment=patterns
 
 import "../../Utils/expect.dart";
 
 String testList(List value) {
-  switch (value) {
-    case const <int>[1, 2]:
-      return "[1, 2]";
-    case const <String>["3", "4"]:
-      return "['3', '4']";
-    case const <double>[]:
-      return "[]";
-    default:
-      return "default";
+  if (value case const [1, 2]) {
+    return "[1, 2]";
+  } else if (value case const ["3", "4"]) {
+    return "['3', '4']";
+  } else if (value case const [true, true]) {
+    return "[true, true]";
+  } else if (value case const []) {
+    return "[]";
+  } else {
+    return "default";
   }
 }
 
 String testMap(Map value) {
-  switch (value) {
-    case const <int, int>{1: 2}:
-      return "{1: 2}";
-    case const <String, num>{'answer': 42}:
-      return "{'answer': 42}";
-    case const <String, Object>{'true': true}:
-      return "{'true': true}";
-    case const <String, int>{}:
-      return "{}";
-    default:
-      return "default";
+  if (value case const {1: 2}) {
+    return "{1: 2}";
+  } else if (value case const {'answer': 42}) {
+    return "{'answer': 42}";
+  } else if (value case const {'true': true}) {
+    return "{'true': true}";
+  } else if (value case const {}) {
+    return "{}";
+  } else {
+    return "default";
   }
 }
 
 String testSet(Set value) {
-  switch (value) {
-    case const <num>{1, 2, 3}:
-      return "{1, 2, 3}";
-    case const <String>{'1', '2', '3'}:
-      return "{'1', '2', '3'}";
-    case const <double>{}:
-      return "{}";
-    default:
-      return "default";
+  if (value case const {1, 2, 3}) {
+    return "{1, 2, 3}";
+  } else if (value case const {'1', '2', '3'}) {
+    return "{'1', '2', '3'}";
+  } else if (value case const {}) {
```

Comment:
I suspect that `{}` could be a `Map<dynamic, dynamic>`.
Cf . https://github.com/dart-lang/language/issues/2637.

For now, we could use `<SomeType>{}` to disambiguate the collection literal, and then we can have a separate test to check that a couple of ambiguous cases using `const {}` as a constantPattern are disambiguated correctly.